ftplugin/elixir.vim

```diff
@@ -52,18 +52,52 @@ silent! setlocal formatoptions-=t formatoptions+=croqlj
 let b:undo_ftplugin = 'setlocal sw< sts< et< isk< com< cms< path< inex< sua< def<'.
       \ '| unlet! b:match_ignorecase b:match_words b:block_begin b:block_end'
 
-function s:projectionist_detect_elixir()
-  if filereadable('mix.exs')
-    let root = simplify(fnamemodify('.', ':p:s?[\/]$??'))
+if !get(g:, 'loaded_projectionist', 1)
+  let g:projectionist_heuristics = get(g:, 'projectionist_heuristics', {})
 
-    let projections = {}
-    let projections['lib/*.ex'] = { 'type': 'main', 'alternate': ['test/{}_test.exs'] }
-    let projections['lib/**/*.ex'] = { 'type': 'main', 'alternate': ['test/{}_test.exs'] }
-    let projections['test/*_test.exs'] = { 'type': 'test', 'alternate': ['lib/{}.ex'] }
-    let projections['test/**/*_test.exs'] = { 'type': 'test', 'alternate': ['lib/{}.ex'] }
-
-    call projectionist#append(root, projections)
-  endif
-endfunc
+  call extend(g:projectionist_heuristics, {
+        \ "&mix.exs":
+        \   {
+        \     'apps/*/mix.exs': { 'type': 'app' },
+        \     'lib/*.ex': {
+        \       'type': 'lib',
+        \       'alternate': 'test/{}_test.exs',
+        \       'template': [
+        \         'defmodule {camelcase|capitalize|dot} do',
+        \         'end'
+        \       ],
+        \     },
+        \     'test/*_test.exs': {
+        \       'type': 'test',
+        \       'alternate': 'lib/{}.ex',
+        \       'dispatch': "mix test test/{}_test.exs`=v:lnum ? ':'.v:lnum : ''`",
+        \       'template': [
+        \         'defmodule {camelcase|capitalize|dot}Test do',
+        \         '  use ExUnit.Case',
+        \         '',
+        \         '  alias {camelcase|capitalize|dot}, as: Subject',
```

Comment:
I do not think that this line is common practice. Even I am questioning it currently.

Reply:
Ture, I will remove the alias and the reference to it.